cmd/pithosctl/healthz.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (C) 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"net/http"
+
+	"github.com/gravitational/pithos-app/internal/pithosctl/pkg/defaults"
+	"github.com/gravitational/pithos-app/internal/pithosctl/pkg/pithos"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	healthzCmd = &cobra.Command{
+		Use:          "healthz",
+		Short:        "Serve pithos health checks",
+		SilenceUsage: true,
+		RunE:         healthz,
+	}
+	accessKey       string
+	secretAccessKey string
+	endpoint        string
+	bucket          string
+	pithosHealthz   *pithos.Healthz
+)
+
+func init() {
+	pithosctlCmd.AddCommand(healthzCmd)
+	healthzCmd.PersistentFlags().StringVar(&accessKey, "access-key", "", "S3 access key")
+	healthzCmd.PersistentFlags().StringVar(&secretAccessKey, "secret-access-key", "", "S3 secret key")
+	healthzCmd.PersistentFlags().StringVar(&endpoint, "endpoint", defaults.HealthzEndpoint, "S3 endpoint address")
+	healthzCmd.PersistentFlags().StringVar(&bucket, "bucket", defaults.HealthzBucket, "S3 bucket name")
+
+	if err := bindFlagEnv(healthzCmd.PersistentFlags()); err != nil {
+		log.WithError(err).Error("Failed to bind environment variables to flags.")
```

Comment:
Is this not supposed to be fatal? What happens when this fails - will the binary end up misconfigured?

Reply:
Empty access/secret keys will fail later in CheckOrSetDefault. Empty bucket/endpoints will be set to defaults.